staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -62,6 +62,42 @@ var (
 		},
 		[]string{"resource"},
 	)
+	watchersCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name: "apiserver_created_watchers",
+			Help: "Counter of watchers created in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	watchersGauge = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name: "apiserver_realtime_watchers",
+			Help: "Gauge of watchers watching in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	cacherEventsCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name: "apiserver_cacher_received_events",
+			Help: "Counter of events received from etcd broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	cacherEventsSentLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Name:    "apiserver_cacher_events_sent_latency_milliseconds",
+			Help:    "Latency buckets of cacher watchers sent events in watchcache broken by resource type",
+			Buckets: metrics.DefBuckets,
+		},
+		[]string{"resource"},
+	)
+	cacherWatcherChannelLength = metrics.NewGaugeVec(
```

Comment:
I would like to better understand the motivation you have for adding this metrics. Can this be actionable?

Reply:
It's useful to check if a cacher is blocking by `dispatchEvents`. 
![image](https://user-images.githubusercontent.com/13895988/67190836-4595f900-f423-11e9-9911-b28993baf31a.png)

Reply:
Same comment as for cacherEventsCounter